src/main/types/SIOP.types.ts

```diff
@@ -90,15 +90,67 @@ export interface AuthenticationResponsePayload extends JWTPayload {
   did: string;
   registration?: DiscoveryMetadataPayload;
   registration_uri?: string;
-  vp_token?: VerifiablePresentationWrapper[];
+  verifiable_presentations?: VerifiablePresentationPayload[];
+  // fixme All of the above is the id token. We need to create a new interface of the above and reference that here as id_token
+  vp_token?: VerifiablePresentationPayload;
   claims?: ResponseClaims;
 }
+/*
+
+export interface OidcClaimJson {
+  essential?: boolean;
+  value?: string;
+  values?: string[];
+}
+
+export interface OidcClaimRequest {
+  [x: string]: null | OidcClaimJson;
+}*/
+
+export interface IdTokenClaimPayload {
+  verifiable_presentations?: PresentationDefinition[];
+  [x: string]: unknown;
+}
+export interface VpTokenClaimPayload {
+  verifiable_presentation?: PresentationDefinition;
+  [x: string]: unknown;
+}
+
+export interface ClaimOpts {
+  presentationDefinitions?: PresentationDefinitionWithLocation[];
+}
+
+export interface ClaimPayload {
+  id_token?: IdTokenClaimPayload;
+  vp_token?: VpTokenClaimPayload;
+  [x: string]: unknown;
+}
+export interface DIDDocument extends DIFDIDDocument {
+  owner?: string;
+  created?: string;
+  updated?: string;
+  proof?: LinkedDataProof;
+}
+
+export interface PresentationDefinitionWithLocation {
```

Comment:
based on [section-11.6.1.1](https://openid.net/specs/openid-connect-4-verifiable-presentations-1_0.html#section-11.6.1.1) the definition (which should be `presentation_definition`) have to be inside the `vp_token` object and therefore `PresentationLocation` can't be an enum